### PR TITLE
Cleanup in various classes

### PR DESCRIPTION
**Pull request by nezihyigitbasi**

### CODE REVIEW

File: bootstrap/src/test/java/io/airlift/bootstrap/TestBootstrap.java [hunk 1]
```diff
@@ -46,14 +45,9 @@ public void testRequiresExplicitBindings()
...
     {
-        Bootstrap bootstrap = new Bootstrap(new Module()
-        {
-            @Override
-            public void configure(Binder binder)
-            {
-                binder.bind(InstanceA.class);
-                binder.bind(InstanceB.class);
-            }
+        Bootstrap bootstrap = new Bootstrap((Module) binder -> {
```
Comment (electrum) [inline bootstrap/src/test/java/io/airlift/bootstrap/TestBootstrap.java]: Is the `(Module)` needed?

File: discovery/src/test/java/io/airlift/discovery/client/TestHttpAnnouncementBinder.java [hunk 2]
```diff
@@ -45,14 +43,9 @@ public void testHttpAnnouncement()
...
-                {
-                    @Override
-                    public void configure(Binder binder)
-                    {
-                        binder.bind(AnnouncementHttpServerInfo.class).toInstance(httpServerInfo);
-                        DiscoveryBinder.discoveryBinder(binder).bindHttpAnnouncement("apple");
-                    }
+                binder -> {
+                    binder.bind(AnnouncementHttpServerInfo.class).toInstance(httpServerInfo);
+                    DiscoveryBinder.discoveryBinder(binder).bindHttpAnnouncement("apple");
```
Comment (electrum) [inline discovery/src/test/java/io/airlift/discovery/client/TestHttpAnnouncementBinder.java]: Static import

File: http-server/src/test/java/io/airlift/http/server/TestHttpServerModule.java [hunk 3]
```diff
@@ -192,18 +176,13 @@ public void testServer()
...
-                        binder.bind(Servlet.class).annotatedWith(TheServlet.class).to(DummyServlet.class);
-                        Multibinder.newSetBinder(binder, Filter.class, TheServlet.class).addBinding().to(DummyFilter.class).in(Scopes.SINGLETON);
-                        httpServerBinder(binder).bindResource("/", "webapp/user").withWelcomeFile("user-welcome.txt");
-                        httpServerBinder(binder).bindResource("/", "webapp/user2");
-                        httpServerBinder(binder).bindResource("path", "webapp/user").withWelcomeFile("user-welcome.txt");
-                        httpServerBinder(binder).bindResource("path", "webapp/user2");
-                    }
+                binder -> {
+                    binder.bind(Servlet.class).annotatedWith(TheServlet.class).to(DummyServlet.class);
+                    Multibinder.newSetBinder(binder, Filter.class, TheServlet.class).addBinding().to(DummyFilter.class).in(Scopes.SINGLETON);
```
Comment (electrum) [inline http-server/src/test/java/io/airlift/http/server/TestHttpServerModule.java]: Static import

File: stats/src/test/java/io/airlift/stats/TestTimeStat.java [hunk 4]
```diff
@@ -90,14 +90,9 @@ public void time()
...
         TimeStat stat = new TimeStat(ticker);
-        stat.time(new Callable<Void>()
-        {
-            @Override
-            public Void call()
-            {
-                ticker.increment(10, TimeUnit.MILLISECONDS);
-                return null;
-            }
+        stat.time((Callable<Void>) () -> {
```
Comment (electrum) [inline stats/src/test/java/io/airlift/stats/TestTimeStat.java]: Is the cast needed? There is only one `time(x)` method

File: discovery/src/main/java/io/airlift/discovery/client/CachingServiceSelector.java [hunk 5]
```diff
@@ -155,14 +155,7 @@ private void scheduleRefresh(Duration delay)
...
         }
-        executor.schedule(new Runnable()
-        {
-            @Override
-            public void run()
-            {
-                refresh();
-            }
-        }, delay.toMillis(), TimeUnit.MILLISECONDS);
+        executor.schedule((Runnable) this::refresh, delay.toMillis(), TimeUnit.MILLISECONDS);
```
Comment (electrum) [inline discovery/src/main/java/io/airlift/discovery/client/CachingServiceSelector.java]: Is the `(Runnable)` needed?

**Comment by nezihyigitbasi** [comment]
Thanks for reviewing. Removed the unnecessary casts & added two new commits for static importing discoveryBinder and newSetBinder.

**Pull request merged**